odoo/api.py

```diff
@@ -720,11 +720,17 @@ def protecting(self, what, records=None):
         protected = self._protected
         try:
             protected.pushmap()
-            what = what if records is None else [(what, records)]
-            for fields, records in what:
-                for field in fields:
-                    ids = protected.get(field, frozenset())
-                    protected[field] = ids.union(records._ids)
+            if records is not None:  # Handle first signature
+                records_by_fields = {field: records._ids for field in what}
+            else:  # Handle second signature
+                records_by_fields = {}
```

Comment:
use a `defaultdict(set)`, this will simplify the code below
you can even use a `list` instead of a `set` (this is going to be faster)

Reply:
> use a defaultdict(set), this will simplify the code below

Agree, I thought this team dislikes `defaultdict`.

> you can even use a list instead of a set (this is going to be faster)

Well, why not (the records has always distinct `ids` anyway, but maybe it will not the case later)